src/manifolds/GeneralizedStiefel.jl

```diff
@@ -0,0 +1,245 @@
+@doc doc"""
+    GeneralizedStiefel{n,k,B,T} <: AbstractEmbeddedManifold{AbstractEmbeddingType}
+
+The Generalized Stiefel manifold consists of all $n\times k$, $n\geq k$ orthonormal
+matrices w.r.t. an arbitrary scalar product with symmetric positive definite matrix
+$B\in R^{n × n}$, i.e.
+
+````math
+\operatorname{St}(n,k,B) = \{ p \in \mathbb F^{n × k} : p^{\mathrm{H}} B p = I_k \},
+````
+
+where $𝔽 ∈ \{ℝ, ℂ\}$,
+$\cdot^{\mathrm{H}}$ denotes the complex conjugate transpose or Hermitian, and
+$I_k \in \mathbb R^{k × k}$ denotes the $k × k$ identity matrix.
+
+
+In the case $B=I_k$ one gets the usual [`Stiefel`](@ref) manifold.
+
+The tangent space at a point $p\in\mathcal M=\operatorname{St}(n,k,B)$ is given by
+
+````math
+T_p\mathcal M = \{ X \in 𝔽^{n × k} : p^{\mathrm{H}}BX + X^{\mathrm{H}}Bp=0_n\},
+````
+where $0_k$ is the $k × k$ zero matrix.
+
+This manifold is modeled as an embedded manifold to the [`Euclidean`](@ref), i.e.
+several functions like the [`zero_tangent_vector`](@ref) are inherited from the embedding.
+
+The manifold is named after
+[Eduard L. Stiefel](https://en.wikipedia.org/wiki/Eduard_Stiefel) (1909–1978).
+
+# Constructor
+    GeneralizedStiefel(n, k, B=I_k, F=ℝ)
+
+Generate the (real-valued) Generalized Stiefel manifold of $n\times k$ dimensional
+orthonormal matrices with scalar product `B`.
+"""
+struct GeneralizedStiefel{n,k,TB<:AbstractMatrix,F} <: AbstractEmbeddedManifold{AbstractEmbeddingType}
```

Comment:
```suggestion
struct GeneralizedStiefel{n,k,TB<:AbstractMatrix,F} <: AbstractEmbeddedManifold{DefaultEmbeddingType}
```
I think it makes sense to have a concrete type here. `DefaultEmbeddingType` would just be a simple struct for default embeddings that are not transparent isometric.

Reply:
Yes that sounds reasonable. Will change that.

Reply:
But it might take some time, I get errors in some dispatches after introducing that type (and a corresponding default iso type).